### DISCUSSION TITLE
[ERROR] ExperimentalWarning: The Fetch API is an experimental feature

### DISCUSSION THREAD

**Discussion by john-zaprite** [vercel/community#1778] [Help] [answered]
I just upgraded my project to _Node.js Version_ **18x** and now I'm getting these errors in my logs:

> ERROR	(node:15) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
> (Use `node --trace-warnings ...` to show where the warning was created)

1. Why does it mention `node:15`?
2. What could be causing this all of a sudden?

**Comment by amyegan** [comment]
Sorry you're running into this error @john-zaprite. Is your repo public, or are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I can't be sure what is causing this error without a little more investigation.

**Reply by john-zaprite** [reply]
Hey Amy, it's not public unfortunately. I will keep digging and see if I can get an MRE for you at some stage.

**Reply by john-zaprite** [reply]
It seems this is a flag that was introduced in Node v17.5:
https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V17.md#add-fetch-api

And is still included in Node v18x:
https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#fetch-experimental

**Comment by DavidForDev** [comment]
I have same error on the vercel project, When I tried to fetch data(from nodejs-graphql) on server side (next js), can anyone help me?

```
export async function getServerSideProps(context) {
  const graphqlSchema = {
    query: `
      query {
        exactlyUser(id: "63c3cf0aea56b467893c92d3") {
          username
        }
      }
  `,
  };

  const fetching = await fetch("http://localhost:5000/", {
    method: "POST",
    body: JSON.stringify(graphqlSchema),
    headers: {
      "Content-Type": "application/json",
    },
  });

  const response = await fetching.json();

  return {
    props: {
      data: response.data,
    },
  };
}

```
![image](https://user-images.githubusercontent.com/87243091/212813450-52e1e960-bb1c-43ef-87ef-99aecaf92416.png)

**Reply by hamzamamadi2001** [reply]
did you find any sloution of this ?

**Reply by DavidForDev** [reply]
> did you find any sloution of this ?

yes I did, nodejs should be deployed, if you want to test it as deployed, you can use ( https://ngrok.com/  ) It is very easy to use

This solution works for me

**Reply by amyegan** [reply]
This seems to be related to the [addition of an experimental fetch API available on the global scope by default](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#fetch-experimental) in Node 18 which can be disabled with the `--no-experimental-fetch` flag

**Reply by DavidForDev** [reply]
> This seems to be related to the [addition of an experimental fetch API available on the global scope by default](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#fetch-experimental) in Node 18 which can be disabled with the `--no-experimental-fetch` flag

@amyegan Thanks for the reply, I'll take that into consideration and next time I will try

**Comment by ryancwalsh** [comment]
Hi @amyegan , I have the same problem as @john-zaprite and @DavidForDev .

My error message is:

```
2023-02-19T18:18:16.165Z	7b8d7013-17a4-40d6-8079-9fc474cceba2	ERROR	(node:8) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
```

I'm using:
```
 "engines": {
    "node": ">=18.12"
  },
```

Everything works locally on my Ubuntu using 18.12.1, which I set via `nvm` to match the exact version that I see Vercel using.

I'll invite you to my project.

**Reply by ryancwalsh** [reply]
@amyegan Is there a way to set Vercel to use Node 18.13.0? https://stackoverflow.com/a/75221620/470749

**Reply by ryancwalsh** [reply]
Importing `node-fetch` (version 2 rather than 3, as hinted here https://stackoverflow.com/a/71808381/470749) seemed to work, but I don't know why it would be necessary.

**Comment by hunain3199** [comment]
> I just upgraded my project to _Node.js Version_ **18x** and now I'm getting these errors in my logs:
> 
> > ERROR	(node:15) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
> > (Use `node --trace-warnings ...` to show where the warning was created)
> 
> 1. Why does it mention `node:15`?
> 2. What could be causing this all of a sudden?

Any one should help me I am facing same problem

**Comment by amyegan** [comment, marked as answer]
Some older dependencies or sub-dependencies may require an earlier version of node. I would take a look at the package.json dependencies and update where possible, or go back to the previous version of Node for the time being
